File: src/main/java/com/brasil/transparente/api/entity/DespesaSimplificada.java
package com.brasil.transparente.api.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "despesa_simplificada")
@Schema(description = "Simplified and reclassified public spending record, grouped according to the project methodology to aggregate comparable social expenditures.")
public class DespesaSimplificada {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Schema(description = "Unique identifier of the simplified spending record")
    private Long despesaSimplificadaId;

    @Schema(description = "Name of the reclassified spending category, such as Health, Education, or Social Security")
    private String name;

    @Schema(description = "Total amount effectively paid for this category, in BRL")
    private double totalValue;

    @Schema(description = "Percentage of the total spending of the federative unit represented by this category")
    private double percentageOfTotal;

    @Schema(description = "Federative Unit identifier to which this spending record belongs. Use 1 for the Brazilian Federal Government")
    private Long unidadeFederativaId;
}

File: src/main/kotlin/com/brasil/transparente/api/entity/DespesaSimplificada.kt
package com.brasil.transparente.api.entity

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "despesa_simplificada")
@Schema(description = "Registro de gasto público simplificado e reclassificado, agrupado de acordo com a metodologia do projeto para agregar despesas comparáveis")
class DespesaSimplificada {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @field:Schema(description = "Identificador único do registro de gasto simplificado")
    var despesaSimplificadaId: Long? = null

    @field:Schema(description = "Nome da categoria de gasto reclassificada")
    var name: String? = null

    @field:Schema(description = "Valor total efetivamente pago para esta categoria, em BRL")
    var totalValue: Double? = null

    @field:Schema(description = "Percentual do total de gastos da unidade pela categoria")
    var percentageOfTotal: Double? = null

    @field:Schema(description = "Identificador da Unidade Federativa à qual este registro de gasto pertence. Use 1 para o Governo Federal")
    var unidadeFederativaId: Long? = null

}